### PR TITLE
Pan updates

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Force-push to fix the build error in secure_partition.c introduced by "core: update ts_store API with user space buffer".

**Comment by jenswi-linaro** [comment]
The checkpatch complaint is only a long line in a print.

File: core/include/kernel/ts_store.h [hunk 1]
```diff
@@ -40,12 +40,17 @@ struct ts_store_ops {
...
 	 * this function.
-	 * @data: pointer to secure memory where the TS bytes should be copied.
-	 * If @data == NULL and @len != 0, the function should just skip @len
-	 * bytes.
+	 * @data_core: pointer to secure memory where the TS bytes should be
+	 *             copied.
+	 * @data_user: pointer to user memory where the TS bytes should be
+	 *             copied.
+	 * At least one of @data_core and @data_user and normally NULL, but
+	 * both are also permitted to be non-null.
```
Comment (jforissier) [inline core/include/kernel/ts_store.h:48]: non-NULL

File: core/include/kernel/ts_store.h [hunk 2]
```diff
@@ -40,12 +40,17 @@ struct ts_store_ops {
...
+	 * If @data_core == NULL, @data_user, and @len != 0, the function
```
Comment (jforissier) [inline core/include/kernel/ts_store.h:49]: If @data_core and @data_user are both NULL, and @len != 0...

**Review by jforissier — COMMENTED** [review]
Reviewing "core: update ts_store API with user space buffer"

**Review by jforissier — COMMENTED** [review]
For commits:


"core: add BB_MEMDUP_USER() and BB_MEMDUP_USER_PRIVATE()"
"core: ldelf: apply finer-grained PAN"
"core: use BB_MEMDUP_USER() where needed"
"core: tee_svc_cryp.c: replace get_used_bits()"

...please apply:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tag applied, update for "core: update ts_store API with user space buffer"

File: core/tee/tadb.c [hunk 3]
```diff
@@ -719,43 +720,58 @@ static TEE_Result ta_load(struct tee_tadb_ta_read *ta)
...
-						  ta->ta_buf + ta->pos +
-							num_bytes, n, b);
-			if (res)
-				break;
-			num_bytes += n;
+		res = tadb_update_payload(ta->ctx, TEE_MODE_DECRYPT,
+					  ta->ta_buf + ta->pos + num_bytes,
+					  n, dst);
+		if (res) {
+			res = TEE_ERROR_SECURITY;
```
Comment (etienne-lms) [inline core/tee/tadb.c:762]: why `TEE_ERROR_SECURITY` here instead of `res`?

File: core/tee/tadb.c [hunk 4]
```diff
@@ -719,43 +720,58 @@ static TEE_Result ta_load(struct tee_tadb_ta_read *ta)
...
 		}
 
-		free(b);
-		if (res)
-			return res;
+		if (buf_user) {
+			res = copy_to_user((uint8_t *)buf_user + num_bytes,
+					   dst, n);
+			if (res) {
+				res = TEE_ERROR_SECURITY;
```
Comment (etienne-lms) [inline core/tee/tadb.c:770]: Here also, why `TEE_ERROR_SECURITY` here instead of `res`?

File: core/include/kernel/ts_store.h [hunk 5]
```diff
@@ -45,8 +45,8 @@ struct ts_store_ops {
 	 * @data_user: pointer to user memory where the TS bytes should be
 	 *             copied.
 	 * At least one of @data_core and @data_user and normally NULL, but
-	 * both are also permitted to be non-null.
-	 * If @data_core == NULL, @data_user, and @len != 0, the function
+	 * both are also permitted to be non-NULL.
+	 * If @data_core == NULL and @data_user, and @len != 0, the function
```
Comment (jforissier) [inline core/include/kernel/ts_store.h:49]: This is still a bit weird, we want both data pointers NULL right? So "If @data_core == NULL and @data_user == NULL and len != 0, the function..."

Comment (jenswi-linaro) [reply core/include/kernel/ts_store.h:49]: Right, I'll update.

File: core/tee/tadb.c (hunk 3 above)
Comment (jenswi-linaro) [reply core/tee/tadb.c:762]: I'll return `res` instead, the same below.

**Comment by jenswi-linaro** [comment]
Tagw applied, update for "core: update ts_store API with user space buffer"

**Comment by jforissier** [comment]
For "core: update ts_store API with user space buffer" with the fixups:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Squashed and tag applied.

**Pull request merged**